pkg/api/v1alpha1/cluster.go

```diff
@@ -463,28 +464,49 @@ func validateEtcdReplicas(clusterConfig *Cluster) error {
 }
 
 func validateNetworking(clusterConfig *Cluster) error {
-	if len(clusterConfig.Spec.ClusterNetwork.Pods.CidrBlocks) <= 0 {
+	clusterNetwork := clusterConfig.Spec.ClusterNetwork
+
+	if len(clusterNetwork.Pods.CidrBlocks) <= 0 {
 		return errors.New("pods CIDR block not specified or empty")
 	}
-	if len(clusterConfig.Spec.ClusterNetwork.Services.CidrBlocks) <= 0 {
+	if len(clusterNetwork.Services.CidrBlocks) <= 0 {
 		return errors.New("services CIDR block not specified or empty")
 	}
-	if len(clusterConfig.Spec.ClusterNetwork.Pods.CidrBlocks) > 1 {
+	if len(clusterNetwork.Pods.CidrBlocks) > 1 {
 		return fmt.Errorf("multiple CIDR blocks for Pods are not yet supported")
 	}
-	if len(clusterConfig.Spec.ClusterNetwork.Services.CidrBlocks) > 1 {
+	if len(clusterNetwork.Services.CidrBlocks) > 1 {
 		return fmt.Errorf("multiple CIDR blocks for Services are not yet supported")
 	}
-	_, _, err := net.ParseCIDR(clusterConfig.Spec.ClusterNetwork.Pods.CidrBlocks[0])
+	_, _, err := net.ParseCIDR(clusterNetwork.Pods.CidrBlocks[0])
 	if err != nil {
-		return fmt.Errorf("invalid CIDR block format for Pods: %s. Please specify a valid CIDR block for pod subnet", clusterConfig.Spec.ClusterNetwork.Pods)
+		return fmt.Errorf("invalid CIDR block format for Pods: %s. Please specify a valid CIDR block for pod subnet", clusterNetwork.Pods)
 	}
-	_, _, err = net.ParseCIDR(clusterConfig.Spec.ClusterNetwork.Services.CidrBlocks[0])
+	_, _, err = net.ParseCIDR(clusterNetwork.Services.CidrBlocks[0])
 	if err != nil {
-		return fmt.Errorf("invalid CIDR block for Services: %s. Please specify a valid CIDR block for service subnet", clusterConfig.Spec.ClusterNetwork.Services)
+		return fmt.Errorf("invalid CIDR block for Services: %s. Please specify a valid CIDR block for service subnet", clusterNetwork.Services)
+	}
+
+	if clusterNetwork.Nodes != nil {
+		podMaskSizeSplit := strings.Split(clusterNetwork.Pods.CidrBlocks[0], "/")
+		podMaskSize, err := strconv.Atoi(podMaskSizeSplit[len(podMaskSizeSplit)-1])
```

Comment:
nit: we could maybe reuse the result of line 481 `net.ParseCIDR(clusterNetwork.Pods.CidrBlocks[0])`